### PR TITLE
Bug 1160615 - Add graph links and graphing to Perf Compare - do not merge

### PR DESCRIPTION
**Pull request by tojonmz**
This (hopefully) fixes Bugzilla bug [1160615](https://bugzilla.mozilla.org/show_bug.cgi?id=1160615).

This adds graph links to the main Perfherder compare page. Based on the way the controllers get assigned in Perherder's html, it seemed I needed to add similar js to `CompareResultsCtrl` that is present in `CompareSubtestResultsCtrl` so the graphs can be accessed.

Current appearance:

comparegraphcurrent

Proposed:

comparegraphproposed

And the result when clicking on 'graph':

graphlinkresult

Maybe there is a better way to do it, but I assumed we wanted to preserve the 1:1 relationship between the current Perf partials and the controllers per https://github.com/mozilla/treeherder/blob/master/ui/js/perfapp.js#L10-L26. Assuming we use this approach I'd still like to re-check what I added to make sure there isn't anything unnecessary, so marking do-not-merge for now.

Tested on OSX 10.10.3:
FF Release **38.0.1**
Chrome Latest Release **42.0.2311.152** (64-bit)

Adding @wlach for initial review and @jmaher for visibility.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/549)

### CODE REVIEW

File: ui/js/compareperf.js [hunk 1]
```diff
@@ -91,8 +91,8 @@ perf.controller('CompareResultsCtrl', [
     function displayResults(rawResultsMap, newRawResultsMap) {
       $scope.compareResults = {};
       $scope.titles = {};
-      window.document.title = ("Comparison between " + $scope.originalRevision + 
-                              " (" + $scope.originalProject.name + ") " + 
+      window.document.title = ("Comparison between " + $scope.originalRevision +
+                              " (" + $scope.originalProject.name + ") " +
```
Comment (tojonmz) [inline ui/js/compareperf.js]: Just noticed unrelated trailing whitespace in the file.

File: ui/js/compareperf.js [hunk 2]
```diff
@@ -119,6 +120,29 @@ perf.controller('CompareResultsCtrl', [
...
+          // Construct the graph link for each row
+          var originalSeries = encodeURIComponent(JSON.stringify(
+                        { project: $scope.originalProject.name,
+                          signature: oldSig,
+                          visible: true}));
+
+          var newSeries = encodeURIComponent(JSON.stringify(
+                        { project: $scope.newProject.name,
+                          signature: newSig,
+                          visible: true}));
```
Comment (tojonmz) [inline ui/js/compareperf.js]: A linter implies there could indentation improvements here but I left them the same as the ones in `CompareSubtestResultsCtrl`.

**Comment by wlach** [comment]
So this looks pretty good, but I wonder if we should create a method called getGraphLink inside PhCompare to create the actual link? As it is there's a bit of duplication with your patch. You could put it in here: https://github.com/mozilla/treeherder/blob/master/ui/js/perf.js#L363

**Comment by tojonmz** [comment]
Ok, will have a look at that.

**Comment by tojonmz** [comment]
Sorry, with all the other things going on I haven't gotten to this.

**Comment by tojonmz** [comment]
I was chatting with @wlach on IRC, and he mentioned the need for this addition is now uncertain. So we'll hold off on this PR pending further usability review of Compare in the coming weeks.

**Comment by tojonmz** [comment]
So chatting again with @wlach and we're uncertain about when/where/how to link from compare->graph, so closing this PR for now unmerged, to avoid it sitting in the PR queue.
